resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -496,7 +481,7 @@ private[yarn] class YarnAllocator(
       executorIdCounter += 1
       val executorHostname = container.getNodeId.getHost
       val containerId = container.getId
-      val executorId = executorIdCounter.toString
+      val executorId = s"$allocatorUuid-$executorIdCounter"
```

Comment:
Are we concerned that this changes Yarn behaviour?

Reply:
(There don't seem to be any associated tests)

Reply:
This change is needed because [this commit](https://github.com/apache/spark/commit/ebe3c7f290929588c822137b8bf27b18fe75393f) relies on numerical increasing ids. That change basically allows `YarnAllocator` to call the driver on start and get the last issued numerical id. After that, it starts incrementing it from that number. It achieves the same goal of generating unique ids for Spark executors. It might be useful to check the discussion in Jira [SPARK-12864](https://issues.apache.org/jira/browse/SPARK-12864).

This class doesn't have tests at all, neither the original commit does. I'm not sure if it worth it to write an artificial test that will verify that `UUID`s are different in different instances of `YarnAllocator`. Covering the whole `YarnAllocator` with tests doesn't seem relevant to this PR.